src/nativeMain/kotlin/commands/Connect.kt

```diff
@@ -0,0 +1,30 @@
+package commands
+
+import ConsoleResultWriter
+import IResultWriter
+import api.IEclairClientBuilder
+import arrow.core.flatMap
+import kotlinx.cli.*
+import kotlinx.coroutines.runBlocking
+import types.NodeInfo
+import types.Serialization
+
+@OptIn(ExperimentalCli::class)
+class ConnectCommand(
+    private val resultWriter: IResultWriter,
+    private val eclairClientBuilder: IEclairClientBuilder
+) : Subcommand(
+    "connect",
+    "Connect to another lightning node. This will perform a connection but no channels will be opened. "
+) {
+    private val password by option(ArgType.String, shortName = "p", description = "Password for the Eclair API (can be found in eclair.conf)").required()
+    private val host by option(ArgType.String, description = "Host URL for the Eclair API (can be found in eclair.conf)").default("http://localhost:8080")
+    private val uri by argument(ArgType.String, description = " This API does not require a target address. Instead, eclair will use one of the addresses published by the remote peer in its node_announcement messages.")
+    override fun execute() = runBlocking {
+        val eclairClient = eclairClientBuilder.build(host, password)
+        val result = eclairClient.connect(uri)
+            .flatMap { apiResponse -> Serialization.decode<NodeInfo>(apiResponse) }
+            .map { decoded -> Serialization.encode(decoded) }
```

Comment:
I tried Implementing this into this [commit](https://github.com/claddyk/eclair-cli/commit/2e981958c9b802912699d294a1142b3a63a10c5e)
But I'm facing multiple TypeErrors, and I'm not able to solve them all.
Please let me know if I am on the right direction here or not?

Reply:
Hey, I tried to do this, but I needed to change the result type to string, which might not be a good idea since rest of the code depends on it.
```
class ConsoleResultWriter : IResultWriter {
    override fun write(result: Either<ApiError, String>) {
        when (result) {
            is Either.Right -> println(result.value)
            is Either.Left -> println("error: ${result.value.message} (code=${result.value.code})")
        }
    }
}
```
This needs to be changed to:
```
class ConsoleResultWriter : IResultWriter {
    override fun write(result: String>) {
//rest of the code
}
```

If I'm in wrong direction, please share some code to refer to.